packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -494,6 +507,14 @@ class _DraggableState<T extends Object> extends State<Draggable<T>> {
     _recognizer = widget.createRecognizer(_startDrag);
   }
 
+  @override
+  void didUpdateWidget(covariant Draggable<T> oldWidget) {
+    if (widget.supportedDevices != oldWidget.supportedDevices) {
+      _recognizer = widget.createRecognizer(_startDrag);
```

Comment:
I believe the old recognizer would need to be disposed?

Reply:
That would actually create problems if there's a gesture ongoing...

Reply:
That's a really good point. I am going to close this for now, I thought this would be an easy pipe-through, but I think the concept needs to be thought out a bit more, especially regarding what to do if the devices change mid-gesture.